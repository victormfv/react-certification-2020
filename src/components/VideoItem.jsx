import React from 'react';

function VideoItem(props) {
  return (
    <div className="w-90 max-w-96 flex-column border rounded-lg m-3 shadow-lg">
      <img
        className="w-full max-h-36 object-cover rounded-t-md"
        src={props.video.thumbnail}
        alt={props.video.title}
      />
      <div className="p-4">
        <h2 className="mb-2">{props.video.title}</h2>
        <p className="text-gray-500">{props.video.description}</p>
      </div>
    </div>
  );
}

export default VideoItem;
