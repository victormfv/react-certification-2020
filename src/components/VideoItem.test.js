import React from 'react';
import VideoItem from "./VideoItem";
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
const video={
    etag: "LRviZfd_p3HDDD2uBk5Qv7zaEQU",
    snippet: {
        title: "title",
        description: "desc",
        thumbnails: {
        medium: {
        url: "url"
        }
        }
}
}
describe("VideoItem Test", () => {
    it('renders correctly', () => {
        const tree = renderer.create(<VideoItem key={video.etag} video={video.snippet} />).toJSON();
        expect(tree).toMatchSnapshot();
      });

      it('Text "title" included', () => {
          render(<VideoItem key={video.etag} video={video.snippet} />)
        const i = screen.getByText(/title/i);
        expect(i).toBeInTheDocument();
      });
});