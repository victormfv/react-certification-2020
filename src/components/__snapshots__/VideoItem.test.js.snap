// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoItem Test renders correctly 1`] = `
<button
  className="w-90 max-w-96 flex-column border rounded-lg m-3 shadow-lg"
  type="button"
>
  <img
    alt="title"
    className="w-full max-h-36 object-cover rounded-t-md"
  />
  <div
    className="p-4"
  >
    <h2
      className="mb-2"
    >
      title
    </h2>
    <p
      className="text-gray-500"
    >
      desc
    </p>
  </div>
</button>
`;
